### DISCUSSION TITLE
Using css variables for responsive support for framer-motion

### DISCUSSION THREAD

**Discussion by adrianlb** [tailwindlabs/tailwindcss#12080] [Help] [answered]
I'm trying to adding animation for my landing page with the use of framer-motion and tailwind.  

Sam Selikoff has put together a great tutorial on "Responsive Framer Motion with Tailwind CSS", https://youtu.be/xSuxsfn13xg?si=GXeKhEe6ym4rtJR6.  The example of this project is here, https://codesandbox.io/p/sandbox/github/samselikoff/2023-08-07-responsive-framer-motion-with-tailwind/tree/main.

Here's my code:

`<motion.img
              initial={{
                opacity: 0,
                x: "var(--x-from, 100px)"
              }}
              whileInView={{
                opacity: 1,
                x: "var(--x-to, 0px)"
              }}
              viewport={{ once: true }}
              transition={{ duration: 0.5, delay: 0.45 }}
              className="w-[500px] max-md:justify mx-auto
                max-sm:[--x-from,0px]
                max-sm:[--x-to,0px]
              "
              src={test}
              alt="" />
`

For some reason, tailwind is not generating media classes for:

- max-sm:[--x-from,0px]

- max-sm:[--x-to,0px]

But in Sam's example project, tailwind is generating the correct media classes.

**Comment by wongjn** [comment]
The classes `max-sm:[--x-from,0px]` and `max-sm:[--x-to,0px]` are invalid Tailwind CSS classes. You may have meant `max-sm:[--x-from:0px]` and `max-sm:[--x-to:0px]` respectively.

**Reply by adrianlb** [reply]
Not sure if I understand what you mean. 

When I inspect the css for the element in Sam's project, here are what I see that tailwind generates:

`@media not all and (min-width: 640px)
.max-sm\:\[--y-to\:0px\] {
    --y-to: 0px;
}`

`@media not all and (min-width: 640px)
.max-sm\:\[--y-from\:16px\] {
    --y-from: 16px;
}`

**Reply by wongjn** [reply, marked as answer]
The classes in Sam's project are `max-sm:[--y-to:0px]` and `max-sm:[--y-from:16px]` for those rules. Notice how the CSS variable name and value are separated by a colon (`:`). However, if we look at your classes, `max-sm:[--x-from,0px]` and `max-sm:[--x-to,0px]`, the CSS variable name and value are separated by a comma (`,`). You may have meant `max-sm:[--x-from:0px]` and `max-sm:[--x-to:0px]` respectively.

**Reply by adrianlb** [reply]
Ah. Thank you for pointing that out! It's working now.
